src/System.Security.Principal.Windows/src/System/Security/Principal/WindowsIdentity.cs

```diff
@@ -211,6 +211,24 @@ private WindowsIdentity(IntPtr userToken, string authType, int isAuthenticated)
 
         private static SafeAccessTokenHandle DuplicateAccessToken(IntPtr accessToken)
         {
+            if (accessToken == IntPtr.Zero)
+            {
+                throw new ArgumentException(SR.Argument_TokenZero);
+            }
+
+            // Find out if the specified token is a valid.
+            uint dwLength = sizeof(uint);
+            if (!Interop.Advapi32.GetTokenInformation(
+                    accessToken,
+                    (uint)TokenInformationClass.TokenType,
+                    IntPtr.Zero,
+                    0,
+                    out dwLength) &&
```

Comment:
It's strange that we're initializing dwLength to sizeof(uint) but then passing it as `out`, which suggests it didn't need to be initialized.  We should either a) just leave it as `uint dwLength;` or b) change the GetTokenInformation declaration to take the last argument as `ref` instead of `out`, based on whatever the Win32 function expects.

Reply:
Yea I just left it the way it was. It's a out-only value an in some places the local is assigned to the current buffer size and is passed in as (..., dwLength, ref dwLength). Anyway it's not incorrect and there are several places to clean up, I'll leave this for a separate PR.